tests/main/nvidia-files/task.yaml

```diff
@@ -0,0 +1,238 @@
+        echo "No driver is available, expecting: skip.d/$SPREAD_SYSTEM/$PACKAGE_VERSION${PACKAGE_SUFFIX:-} to say: no-driver"
+        test "$(cat "skip.d/$SPREAD_SYSTEM/$PACKAGE_VERSION${PACKAGE_SUFFIX:-}")" = "no-driver"
+        exit 0
+    fi
+
+    # Second, some drivers are only transitional support packages that don't
+    # actually ship any files.
+    if apt-cache show nvidia-driver-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}" | grep -i transitional; then
+        echo "Transitional driver is in use, expecting: skip.d/$SPREAD_SYSTEM/$PACKAGE_VERSION${PACKAGE_SUFFIX:-} to say: transitional-driver"
+        test "$(cat "skip.d/$SPREAD_SYSTEM/$PACKAGE_VERSION${PACKAGE_SUFFIX:-}")" = "transitional-driver"
+        exit 0
+    fi
+
+    # Third, some combinations are really buggy.
+    case "$SPREAD_SYSTEM/$PACKAGE_VERSION/${PACKAGE_SUFFIX:-}" in
+    ubuntu-18.04-64/515/* | ubuntu-18.04-64/390/* | ubuntu-2[02].04-64/390/*)
+        # This fails with:
+        # + exec /snap/test-snapd-nvidia/x1/bin/dlopen-tool.64
+        # /var/lib/snapd/lib/gl/libEGL_nvidia.so.0
+        # /var/lib/snapd/lib/gl/libEGL_nvidia.so.390.157
+        # /var/lib/snapd/lib/gl/libGLESv1_CM_nvidia.so.1
+        # /var/lib/snapd/lib/gl/libGLESv1_CM_nvidia.so.390.157
+        # /var/lib/snapd/lib/gl/libGLESv2_nvidia.so.2
+        # /var/lib/snapd/lib/gl/libGLESv2_nvidia.so.390.157
+        # /var/lib/snapd/lib/gl/libGLX_nvidia.so.0
+        # /var/lib/snapd/lib/gl/libGLX_nvidia.so.390.157
+        # /var/lib/snapd/lib/gl/libcuda.so /var/lib/snapd/lib/gl/libcuda.so.1
+        # /var/lib/snapd/lib/gl/libcuda.so.390.157
+        # /var/lib/snapd/lib/gl/libnvcuvid.so
+        # /var/lib/snapd/lib/gl/libnvcuvid.so.1
+        # /var/lib/snapd/lib/gl/libnvcuvid.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-compiler.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-eglcore.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-encode.so
+        # /var/lib/snapd/lib/gl/libnvidia-encode.so.1
+        # /var/lib/snapd/lib/gl/libnvidia-encode.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-fatbinaryloader.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-fbc.so
+        # /var/lib/snapd/lib/gl/libnvidia-fbc.so.1
+        # /var/lib/snapd/lib/gl/libnvidia-fbc.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-glcore.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-glsi.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-ml.so
+        # /var/lib/snapd/lib/gl/libnvidia-ml.so.1
+        # /var/lib/snapd/lib/gl/libnvidia-ml.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-opencl.so.1
+        # /var/lib/snapd/lib/gl/libnvidia-opencl.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-ptxjitcompiler.so
+        # /var/lib/snapd/lib/gl/libnvidia-ptxjitcompiler.so.1
+        # /var/lib/snapd/lib/gl/libnvidia-ptxjitcompiler.so.390.157
+        # /var/lib/snapd/lib/gl/libnvidia-tls.so.390.157
+        # *** stack smashing detected ***: terminated
```

Comment:
I do not think it is necessary to have all these libraries in the comment, with something like:
```
        # This fails with:
        # + exec /snap/test-snapd-nvidia/x1/bin/dlopen-tool.64
        # /var/lib/snapd/lib/gl/libEGL_nvidia.so.0
        # ...
        # /var/lib/snapd/lib/gl/libnvidia-tls.so.390.157
        # *** stack smashing detected ***: terminated
```
things should be clear enough.

Reply:
agreed, fixed